frontend/public/components/pod-logs.jsx

```diff
@@ -1,194 +1,79 @@
 
-const dataHasFailureMsg = (data) => {
-  return _.includes(data, '"status": "Failure"');
+const containerIsRunning = (container) => {
+  return !_.isEmpty(_.get(container, 'status.running'));
 };
 
-const dataHasHTML = (data) => {
-  return _.includes(data, '<html') || _.includes(data, '<HTML');
+// Component to container dropdown or conatiner name if only one container in pod.
+const ContainerDropdown = ({currentContainer, containers, kind, onChange}) => {
+  const resourceName = (container) => <ResourceName name={container.name || <LoadingInline />} kind={kind} />;
+  const dropdownItems = _.mapValues(containers, resourceName);
+  return <Dropdown className="btn-group" items={dropdownItems} title={resourceName(currentContainer)} onChange={onChange} />;
 };
 
-export class PodLogs extends SafetyFirst {
+export class PodLogs extends React.Component {
   constructor(props) {
     super(props);
-
-    this._buffer = lineBuffer(1000);
-    this._loadTime = Date.now();
-    this._pendingReload = null;
-
-    this._touchLoadTimeState = _.throttle(this._touchLoadTimeState, 100);
     this._selectContainer = this._selectContainer.bind(this);
-    this._updateLogState = this._updateLogState.bind(this);
-    this._toggleLogState = this._toggleLogState.bind(this);
 
     this.state = {
-      containerNames: [],
-      currentContainer: '',
-      logURL: '',
-      loadTime: 0,
-      logState: 'loading'
+      containers: [],
+      currentContainer: {},
+      eof: false
     };
-    this.state = _.defaults({}, this._initialState(), this.state);
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this._beginStreaming();
-  }
-
-  componentWillReceiveProps(nextProps) {
-    this.setState(this._initialState(nextProps.obj));
   }
 
-  componentWillUnmount() {
-    super.componentWillUnmount();
-    this._endStreaming();
-  }
-
-  _initialState(obj = this.props.obj) {
+  static getDerivedStateFromProps(nextProps, prevState) {
     const newState = {};
-
-    const containers = _.get(obj, 'spec.containers', []);
-    newState.containerNames = _.map(containers, 'name');
-
-    if (!this.state.currentContainer && newState.containerNames.length > 0) {
-      newState.currentContainer = newState.containerNames[0];
-    }
-
-    const currentContainer = newState.currentContainer || this.state.currentContainer;
-    newState.logURL = this._logURL(obj, currentContainer);
-
-    return newState;
-  }
-
-  _logURL(obj, currentContainer) {
-    return resourceURL(PodModel, {
-      ns: _.get(obj, 'metadata.namespace'),
-      name: _.get(obj, 'metadata.name'),
-      path: 'log',
-      queryParams: {
-        container: currentContainer,
-        follow: 'true',
-        tailLines: this._buffer.maxSize
-      }
+    const containers = _.get(nextProps.obj, 'status.containerStatuses', []);
+    newState.containers = _.map(containers, (container) => {
+      return {
+        name: container.name,
+        status: container.state
```

Comment:
The only usage of `status` is in `containerIsRunning`.  It looks like that value is always turned into `!eof`.  Why not store the actual value of `eof`?